wire-ios/Wire-iOS/Sources/AppRootRouter.swift

```diff
@@ -345,25 +346,29 @@ extension AppRootRouter {
                                completion: completion)
     }
 
+    @MainActor
     private func showAuthenticated(
         userSession: UserSession,
         completion: @escaping () -> Void
     ) {
-        guard
-            let selectedAccount = SessionManager.shared?.accountManager.selectedAccount,
-            let authenticatedRouter = buildAuthenticatedRouter(
-                account: selectedAccount,
-                userSession: userSession
-            )
-        else {
-            completion()
-            return
-        }
-
-        self.authenticatedRouter = authenticatedRouter
+        Task {
```

Comment:
I wouldn't have expected changes to AppRootRouter.swift. Is this really required for your feature?

Reply:
Yes. As the usecases that we set as dependencies are fetched asynchronously. this has led to this change.